iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -355,15 +291,13 @@ class expected<ErrorType>
     ///         std::cout << "we are successful!" << std::endl;
     ///     })
     /// @endcode
-    [[deprecated]] expected& on_success(const cxx::function_ref<void()>& callable) noexcept;
     expected& and_then(const cxx::function_ref<void()>& callable) noexcept;
 
   private:
     expected(variant<ErrorType>&& store, const bool hasError) noexcept;
-
-  private:
     variant<ErrorType> m_store;
     bool m_hasError;
+    bool m_isInitialized{true};
```

Comment:
This member should be renamed to `m_hasUndefinedState` to be consistent with the `has_undefined_state` method and in the code the developer would see immediately what is going on in there.